sklearn/decomposition/tests/test_incremental_pca.py

```diff
@@ -24,16 +27,43 @@ def test_incremental_pca():
     X_transformed = ipca.fit_transform(X)
 
     np.testing.assert_equal(X_transformed.shape, (X.shape[0], 2))
-    assert_almost_equal(ipca.explained_variance_ratio_.sum(),
-                        pca.explained_variance_ratio_.sum(), 1)
+    np.testing.assert_allclose(ipca.explained_variance_ratio_.sum(),
+                               pca.explained_variance_ratio_.sum(), atol=1e-3)
+
+    for n_components in [1, 2, X.shape[1]]:
+        ipca = IncrementalPCA(n_components, batch_size=batch_size)
+        ipca.fit(X)
+        cov = ipca.get_covariance()
+        precision = ipca.get_precision()
+        np.testing.assert_allclose(np.dot(cov, precision),
+                                   np.eye(X.shape[1]), atol=1e-13)
+
+
+@pytest.mark.parametrize(
+    "matrix_class",
+    [sparse.csc_matrix, sparse.csr_matrix, sparse.lil_matrix])
+def test_incremental_pca_sparse(matrix_class):
+    # Incremental PCA on sparse arrays.
+    X = iris.data
+    pca = PCA(n_components=2)
+    pca.fit_transform(X)
+    X = matrix_class(X)
+    batch_size = X.shape[0] // 3
+    ipca = IncrementalPCA(n_components=2, batch_size=batch_size)
+
+    X_transformed = ipca.fit_transform(X)
+
+    np.testing.assert_equal(X_transformed.shape, (X.shape[0], 2))
```

Comment:
for equality tests of int or lists/tuples of int you can just use `assert a == b`